guake/callbacks.py

```diff
@@ -62,12 +62,19 @@ def on_search_on_web(self, *args):
             query = clipboard.wait_for_text()
             query = quote_plus(query)
 
-            if query:
-                # put the query at the end of the url and https
-                search_url = (
-                    "https://www." + ENGINES[self.settings.general.get_int("search-engine")] + query
-                )
-                Gtk.show_uri(self.window.get_screen(), search_url, get_server_time(self.window))
+            # nothing selected
+            if not query:
+                return
+
+            selected = self.settings.general.get_int("search-engine")
+            # if custom search is selected, get the engine from the 'custom-search-engine' setting
+            if selected == 4:
```

Comment:
The check for the custom option should be checking to see if it's the last index. Hardcoding to 4 can generate unexpected bugs if the list of preset search engines changes length in the future.